### PR TITLE
force __future__.annotations with isort

### PR DESCRIPTION
**Pull request by graingert**
Closes #xxxx

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ repos:
         args:
           - --target-version=py38
   - repo: https://gitlab.com/pycqa/flake8
-    rev: 4.0.1
+    rev: 3.9.2
```
Comment (crusaderky) [inline .pre-commit-config.yaml:27]: Why the downgrade?

File: distributed/__init__.py [hunk 2]
```diff
@@ -1,5 +1,10 @@
-from distributed import config  # isort:skip; load distributed configuration first
-from distributed import widgets  # isort:skip; load distributed widgets second
+from __future__ import annotations
+
+# isort: off
```
Comment (graingert) [inline distributed/__init__.py:3]: `isort: skip` doesn't work unless the block of skipped imports is at the top or bottom of all the imports

Comment (graingert) [reply distributed/__init__.py:3]: https://github.com/PyCQA/isort/issues/1337

File: .pre-commit-config.yaml (hunk 1 above)
Comment (graingert) [reply .pre-commit-config.yaml:27]: oh that's odd I ran `pre-commit autoupgrade`

Comment (graingert) [reply .pre-commit-config.yaml:27]: oh it's gitlab

File: distributed/diagnostics/cluster_dump.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
-from typing import Any, Collection, Dict, Literal
+from __future__ import annotations
+
+from typing import Any, Collection, Literal
```
Comment (crusaderky) [inline distributed/diagnostics/cluster_dump.py:3]:
```suggestion
from collections.abc import Collection
from typing import Any, Literal
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (crusaderky) [inline .pre-commit-config.yaml:27]:
```suggestion
    rev: 4.0.1
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -24,7 +24,7 @@ repos:
         args:
           - --target-version=py38
   - repo: https://gitlab.com/pycqa/flake8
```
Comment (crusaderky) [inline .pre-commit-config.yaml:26]:
```suggestion
  - repo: https://github.com/pycqa/flake8
```

**Review by crusaderky — APPROVED** [review]

**Pull request merged**